hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/StoreFileWriter.java

```diff
@@ -235,10 +210,10 @@ private byte[] toCompactionEventTrackerBytes(Collection<HStoreFile> storeFiles)
    */
   public void appendMetadata(final long maxSequenceId, final boolean majorCompaction,
     final long mobCellsCount) throws IOException {
-    writer.appendFileInfo(MAX_SEQ_ID_KEY, Bytes.toBytes(maxSequenceId));
-    writer.appendFileInfo(MAJOR_COMPACTION_KEY, Bytes.toBytes(majorCompaction));
-    writer.appendFileInfo(MOB_CELLS_COUNT, Bytes.toBytes(mobCellsCount));
-    appendTrackedTimestampsToMetadata();
+    liveFileWriter.appendMetadata(maxSequenceId, majorCompaction, mobCellsCount);
```

Comment:
Is this the correct behavior? We should track mobCellsCount for the two writers separately?

Reply:
I did not change the behavior. This change is due to refactoring the code. We do not support dual file writing for mob yet as mentioned in the previous comment.